### ISSUE TITLE
Switch conjugation back to base view

### ISSUE THREAD

**Issue by andrewtavis** [scribe-org/Scribe-iOS#168]
### Terms

- [X] I have searched all [open bug reports](https://github.com/scribe-org/Scribe-iOS/issues?q=is%3Aopen+is%3Aissue+label%3Abug)
- [X] I agree to follow Scribe-iOS' [Code of Conduct](https://github.com/scribe-org/Scribe-iOS/blob/main/.github/CODE_OF_CONDUCT.md)

### Behavior

This is a very simple issue. Currently the conjugation state is not resetting for the conjugation command, which means that if a user conjugated a verb in past tense, then the next time they use the command past tense will be the first one shown. Rather what we'd like is for conjugation to always start with present tense, or whatever the most simple conjugation forms are.

What needs to happen is that the conjugation state needs to be reset after the command is executed, which could go in the block for `case "return"` of `executeKeyActions` of [KeyboardsBase/KeyboardViewController.swift](https://github.com/scribe-org/Scribe-iOS/blob/main/Keyboards/KeyboardsBase/KeyboardViewController.swift).

### Device type

iPhone and iPad

### iOS version

I assume this is happening for all versions.

**Comment by andrewtavis** [comment]
This seems to only be affecting certain keyboards and not others. The German keyboard does switch back, for instance, but the Spanish one does not.

**Comment by andrewtavis** [comment]
@SaurabhJamadagni, this issue deals with resetting the conjugation state before the interface is loaded. Looking at it, I only did it for German and not any of the other ones (I'm normally testing on the German keyboard). The conjugation state is reset for German at this place in the codes, which is right when the conjugation button is pressed:

https://github.com/scribe-org/Scribe-iOS/blob/076650f8cd1f36c811636a226d4dbe83340705eb/Keyboards/KeyboardsBase/KeyboardViewController.swift#L1272

The line `deConjugationState = .indicativePresent` resets german as German's base conjugation is this one as defined below:

https://github.com/scribe-org/Scribe-iOS/blob/076650f8cd1f36c811636a226d4dbe83340705eb/Keyboards/LanguageKeyboards/German/DECommandVariables.swift#L23

Basically we want to do the same for the baseline conjugation states for the other languages, which are all found in the corresponding `CommandVariables.swift` files. Six lines are needed - one for each missing conjugation reset - and maybe a space afterwards would also help with readability :)

**Comment by SaurabhJamadagni** [comment]
Thank you very much @andrewtavis! I'll get started with this immediately.

**Comment by SaurabhJamadagni** [comment]
Hi @andrewtavis! I went ahead to make the required changes to the six files right now, but the code lines that are needed are already there in each file. Do you think there can be some other problem that is causing the conjugation to not revert back? I will brainstorm about any other possibilities that might be causing this as well.

**Comment by andrewtavis** [comment]
@SaurabhJamadagni, the six files are fine, but they need to be referenced to find the needed lines to be added to: https://github.com/scribe-org/Scribe-iOS/blob/076650f8cd1f36c811636a226d4dbe83340705eb/Keyboards/KeyboardsBase/KeyboardViewController.swift#L1272

Of the seven languages, German is the only one that has a conjugation reset line in the code referenced above. Below `deConjugationState = .indicativePresent`, we also need a line `frConjugationState = .indicativePresent` as that's the base conjugation as you've seen here: https://github.com/scribe-org/Scribe-iOS/blob/5e7da847c9994b060dde7f1494b0e6c355263229/Keyboards/LanguageKeyboards/French/FRCommandVaribles.swift#L23

So along with `frConjugationState = .indicativePresent` we also need to find the base conjugation states of Spanish, etc and add them too :) Let me know if it's still confusing, and sorry my directions weren't clear enough!

**Comment by andrewtavis** [comment]
@SaurabhJamadagni, assigning you to this one and the other we're discussing 😊

**Comment by SaurabhJamadagni** [comment]
> So along with frConjugationState = .indicativePresent we also need to find the base conjugation states of Spanish, etc and add them too :) Let me know if it's still confusing, and sorry my directions weren't clear enough!

Oh I understand! That was very silly of me. Your directions were perfectly clear! My bad 😅

**Comment by andrewtavis** [comment]
@SaurabhJamadagni, just checked it on the emulator for all the new languages, and it's working just fine :D Thank you for such a helpful contribution 😊

As mentioned in #173, this will be included in version 1.3.8 with a data update and the new App Store images 🚀 I'm going to eat now and have a presentation on Scribe for some universities later, but the update should be sent later on tonight :) Releases are accepted within 48 hours, so it should be live soon!

Thanks again so much!

**Comment by SaurabhJamadagni** [comment]
Wouldn't have been possible without your immense help @andrewtavis! Thank you so much for every explanation you provided. My understanding of the codebase is only going to get better from here to help you with larger contributions. I look forward to the update on the app store! It is thrilling to know that my contribution will be a part of an app on the app store!

Do you use any other communication channels where we could converse? I would love to know more about your work. I have much to learn. Once again, thank you for all the help you provided. I look forward to my future contributions with Scribe-iOS!

**Comment by andrewtavis** [comment]
You're very very welcome, @SaurabhJamadagni! Very much looking forward to larger contributions and working on more features with you down the road! You can check what the App Store will look like now in #167 😊 Might need to be cleaned up a bit, but the designer and I are very happy with how it turned out 😊😊

As far as other communication channels, you're welcome to reach out at my email that's on my profile. I'm honestly not really a social media user. As you work further I'd be happy to discuss the project and other things more in depth with you, and in that case we could schedule a call or something like that :)

Wishing you all the best!
